### PR TITLE
FED-4495 System sx migrator

### PR DESCRIPTION
**Pull request by greglittlefield-wf**
## Motivation
MUI v6 deprecated system props on its components, instructing consumers to migrate to `sx` instead: https://mui.com/material-ui/migration/migrating-from-deprecated-apis/#system-props

We have some MUI Dart wrappers that should be migrated, that their existing codemod won't work for.

## Changes
- Add `mui_system_props_migration` codemod executable, which migrates system props to `sx`
    - Detects usages of OverReact props that wrap MUI components regardless of where they're defined (duck-typing them by looking for `sx` and deprecated system props)
        - This allows us to run tests in CI in this OSS repo, and had the bonus of automatically handling the `Link` component I originally missed
    - Handles edge cases of existing `sx` props and prop forwarding, merging them while preserving existing behavior as best as possible
        - Most cases are handled automatically, but some require manual intervention and will get a FIXME comment with context on how to address
- Add tests

#### Example migration
Before (system props):
```dart
(Box()..m = 2)();
(Box()
  ..m = 2
  ..sx = {'color': '#f00'}
)()
(Box()
  ..m = 2
  ..addProps(props.getPropsToForward())
)()
```

After (all system props migrated to SX):
```dart
(Box()..sx = {'m': 2})()
(Box()..sx = {
  'm': 2,
  'color': '#f00'
})()
(Box()
  ..addProps(props.getPropsToForward())
  ..sx = {
    'm': 2,
    ...?props.sx,
  }
)()
```

## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
        - Verify tests run in CI and pass
        - Run the codemod locally on a few repos to make sure the codemod doesn't crash, and verify output looks good and FIXME comments show up where expected and make sense
            ```
            dart pub global activate --source=git --git-ref=system-sx-migrator https://github.com/Workiva/over_react_codemod
            mui_system_props_migration --yes-to-all
            ddev format
            ```
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Frontend Frameworks Design member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/over_react_codemod/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/over_react_codemod/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

File: test/mui_suggestors/system_props_to_sx_migrator_test.dart [hunk 1]
```diff
@@ -0,0 +1,1244 @@
...
+    String withImports(String source) => '''
+      //@dart=2.19
+      import 'package:over_react/over_react.dart';
+      import ${jsonEncode(muiUri.toString())};
+      $source
+    ''';
+
+    const sxPrecedenceFixme =
+        '// FIXME(mui_system_props_migration) - Previously, it was possible for forwarded system props to overwrite these migrated styles, but not anymore since sx takes precedence over any system props.'
+        '\n //  Double-check that this new behavior is okay.';
```
Comment (kealjones-wk) [inline test/mui_suggestors/system_props_to_sx_migrator_test.dart]: This is a hard thing to consider... so this is only really an issue when the usage of the component hasn't been migrated to `sx` and is still using system props... I was gonna say what if we made a utility like `systemPropsToSx(props)` that would just like grab any system props name and move it into the props.sx object... but like that seems like overkill and like it is a pretty unlikely case.

**Review by kealjones-wk — DISMISSED** [review]
+10

Comments about possible changes are totally optional... seems like it covers everything really well. Nice work.

**Review by kealjones-wk — APPROVED** [review]
+10

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM
